ddtrace/internal/core.py

```diff
@@ -20,6 +20,102 @@
 except ImportError:
     import ddtrace.vendor.contextvars as contextvars  # type: ignore
 
+"""
+This file implements the Core API, the abstraction layer between Integrations and Product code.
+The Core API enables two primary use cases: maintaining a tree of ``ExecutionContext`` objects
+and dispatching events.
+
+When using the Core API, keep concerns separate between Products and Integrations. Integrations
+should not contain any code that references Products (Tracing, AppSec, Spans, WAF, Debugging, et cetera)
+and Product code should never reference the library being integrated with.
```

Comment:
> Product code should never reference the library being integrated with

What do you mean by this? Should it be clarified? Like, the Tracing code should never import the `flask` module?